app/src/main/java/com/parking_system_kotlin/mvp/view/ParkingSpacesSettingView.kt

```diff
@@ -0,0 +1,27 @@
+package com.parkingsystemkotlin.mvp.view
+
+import android.content.Context
+import android.widget.Toast
+import com.parkingsystemkotlin.R
+import com.parkingsystemkotlin.fragments.SpacesSettingDialogFragment
+import com.parkingsystemkotlin.listeners.ListenerSetParkingDialogFragment
+import com.parkingsystemkotlin.mvp.contracts.ParkingSpacesSettingContract
+import com.parkingsystemkotlin.mvp.view.base.FragmentView
+
+class ParkingSpacesSettingView(fragment: SpacesSettingDialogFragment) : FragmentView(fragment), ParkingSpacesSettingContract.ParkingSpacesSettingDialogFragmentView{
+
+    override fun showParkingLotsAvailable(parkingLots: String, listenerSetParkingDialogFragment: ListenerSetParkingDialogFragment) {
+        val fragment: SpacesSettingDialogFragment = getFragment() as SpacesSettingDialogFragment
+        if (fragment!=null){
+            fragment.dismiss()
+        }
+        listenerSetParkingDialogFragment.listenFreeSpaces(parkingLots)
+    }
+
+    override fun showInvalidValue() {
+        val context: Context? = getContext()
+        if (context != null){
```

Comment:
instead of this you can use context?.let { <the toast here>}

Reply:
done